WalletWasabi/Helpers/PowerSaving/LinuxInhibitorTask.cs

```diff
@@ -0,0 +1,310 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+using WalletWasabi.Microservices;
+
+namespace WalletWasabi.Helpers.PowerSaving
+{
+	/// <summary><c>systemd-inhibitor</c> API wrapper.</summary>
+	/// <remarks>Only works on Linux machines that use systemd.</remarks>
+	/// <seealso href="https://www.freedesktop.org/wiki/Software/systemd/inhibit/"/>
+	public class LinuxInhibitorTask : IPowerSavingInhibitorTask
+	{
+		/// <remarks>Guarded by <see cref="StateLock"/>.</remarks>
+		private bool _isDone;
+
+		/// <remarks>Use the constructor only in tests.</remarks>
+		internal LinuxInhibitorTask(InhibitWhat what, TimeSpan period, string reason, ProcessAsync process)
+		{
+			What = what;
+			BasePeriod = period;
+			Reason = reason;
+			Process = process;
+			Cts = new CancellationTokenSource(period);
+
+			Task = WaitAsync();
```

Comment:
It is an acceptable convention to have Start and Stop for classes that are running something under the hood (in this case a process). OR Stop is dispose?

Reply:
I'm not sure I understand properly but the reasoning behind this is: You start an inhibitor task and then you stop it. 

Do you have better a name/convention in mind?